src/games/connect_four.rs

```diff
@@ -228,14 +242,246 @@ impl ConnectFour {
 		self.cells[offset] = Player::Unset;
 	}
 
+	fn evaluate_window(&self, player: Player, i: &[usize; 4]) -> i32 {
+		const MASK_EMPTY: Simd<i8, 4> = i8x4::from_array([Player::Unset as i8; 4]);
+		const MASK_HUMAN: Simd<i8, 4> = i8x4::from_array([Player::Human as i8; 4]);
+		const MASK_MACHINE: Simd<i8, 4> = i8x4::from_array([Player::Machine as i8; 4]);
+
+		let v = i8x4::from_array(i.map(|x| unsafe { (*self.cells.get_unchecked(x)).try_into().unwrap_unchecked() }));
+		let player_mask = if player == Player::Human { MASK_HUMAN } else { MASK_MACHINE };
+
+		let empty_pieces = bitmask_to_count(v.simd_eq(MASK_EMPTY));
+		let player_pieces = bitmask_to_count(v.simd_eq(player_mask));
+
+		// The sum of empty and player pieces must be less than or equal to 4:
+		debug_assert!(empty_pieces + player_pieces <= 4);
+		// Winning moves will never call this function:
+		debug_assert_ne!(player_pieces, 4);
+		// Losing moves will never call this function:
+		debug_assert_ne!(player_pieces + empty_pieces, 0);
+
+		let mask: u8 = (player_pieces << 3) | empty_pieces;
+		match mask {
+			// 4 player pieces: (winning move, cannot happen here)
+			0b100_000 => unsafe { unreachable_unchecked() },
```

Comment:
use octal to make these more readable, like this
```suggestion
			0o4_0 => unsafe { unreachable_unchecked() },
```

Reply:
You saviour, I was kinda "darn, I wish 3-bit literals were a thing" and never occurred to me that octals are literally that 😅